libavfilter/dnn/dnn_backend_native_layer_conv2d.c

```diff
@@ -152,35 +211,35 @@ static void * dnn_execute_layer_conv2d_thread(void *threadarg)
 
     av_assert0(channel == conv_params->input_num);
 
+    struct execute_param exe_param;
+    exe_param.thread_start = thread_start;
+    exe_param.thread_end = thread_end;
+    exe_param.input = input;
+    exe_param.output = output;
+    exe_param.kernel = conv_params->kernel;
+    exe_param.input_num = conv_params->input_num;
+    exe_param.output_num = conv_params->output_num;
+    exe_param.kernel_size = conv_params->kernel_size;
+    exe_param.padding_method = conv_params->padding_method;
+    exe_param.dilation = conv_params->dilation;
+    exe_param.pad_size = pad_size;
+    exe_param.width = width;
+    exe_param.height = height;
+    exe_param.radius = radius;
+    exe_param.src_linesize = src_linesize;
+    exe_param.filter_size = filter_size;
+    exe_param.filter_linesize = filter_linesize;
+
+    ff_dnn_execute_layer_conv2d_c(&exe_param);
+
+    output = output_operand->data;
+    output += (conv_params->output_num) * (width - 2 * pad_size) * (thread_start - pad_size);
     for (int y = thread_start; y < thread_end; ++y) {
         for (int x = pad_size; x < width - pad_size; ++x) {
             for (int n_filter = 0; n_filter < conv_params->output_num; ++n_filter) {
                 if (conv_params->has_bias)
-                    output[n_filter] = conv_params->biases[n_filter];
-                else
-                    output[n_filter] = 0.f;
-
-                for (int ch = 0; ch < conv_params->input_num; ++ch) {
-                    for (int kernel_y = 0; kernel_y < conv_params->kernel_size; ++kernel_y) {
-                        for (int kernel_x = 0; kernel_x < conv_params->kernel_size; ++kernel_x) {
-                            float input_pel;
-                            if (conv_params->padding_method == SAME_CLAMP_TO_EDGE) {
-                                int y_pos = CLAMP_TO_EDGE(y + (kernel_y - radius) * conv_params->dilation, height);
-                                int x_pos = CLAMP_TO_EDGE(x + (kernel_x - radius) * conv_params->dilation, width);
-                                input_pel = input[y_pos * src_linesize + x_pos * conv_params->input_num + ch];
-                            } else {
-                                int y_pos = y + (kernel_y - radius) * conv_params->dilation;
-                                int x_pos = x + (kernel_x - radius) * conv_params->dilation;
-                                input_pel = (x_pos < 0 || x_pos >= width || y_pos < 0 || y_pos >= height) ? 0.0 :
-                                                   input[y_pos * src_linesize + x_pos * conv_params->input_num + ch];
-                            }
-
+                    output[n_filter] += conv_params->biases[n_filter];
```

Comment:
why we don't put this into _c function.

Reply:
This should be put into _c function and can be used as initial value of output.